### ISSUE TITLE
Throttling Breaks when Deleting Processing Jobs

### ISSUE THREAD

**Issue by nickmkk** [HangfireIO/Hangfire#1971]
I recently found an issue where jobs go straight to scheduled after using the delete feature on the dashboard.  The job in question is using a semaphore to limit concurrent execution to 5 at a time.  After deleting 5 jobs that were in the processing state all new jobs go straight to the scheduled state.  I looked at the hangfire redis db and found what appears to be the semaphore for my job and once removing the 5 entries shown in the attached photo jobs began processing again.  It appears deleting a processing job does not properly release the semaphore.
![semaphore](https://user-images.githubusercontent.com/10885165/142294125-65485651-ae75-4f70-ac67-1e4c6ea2b48a.png)

**Comment by odinserj** [comment]
Yes, they should reference the same code – extension filter may significantly change the processing behaviour so it's important to include them even into Dashboard UI. Most likely I will create microservice-oriented extension after releasing Hangfire 1.8.0 that will allow to detach monitoring and execution from the specific code with some limitations, but currently it's possible only for methods that's reside in some shared assembly.

**Comment by nickmkk** [comment]
I see, as long as we don't take any actions like requeueing jobs or deleting processing jobs to cancel them it shouldn't cause a problem right?  We may just need to avoid job cancellation through the dashboard.

**Comment by odinserj** [comment]
Yes, you can even activate the `DashboardOptions.IsReadOnlyFunc` option to always return `true` to ensure actions are inaccessible.

**Comment by nickmkk** [comment]
I had considered that but I don't think we can delete failed jobs when that is active.  I will probably just limit write access to a small group.  Thanks for your help
